src/Glpi/Asset/AssetDefinitionManager.php

```diff
@@ -436,4 +444,155 @@ final class {$definition->getAssetTypeClassName(false)} extends AssetType {
         $reflected_class = new ReflectionClass($definition->getAssetTypeClassName());
         $reflected_class->setStaticPropertyValue('definition', $definition);
     }
+
+    private function loadConcreteModelDictionaryClass(AssetDefinition $definition): void
+    {
+        $model_class = $definition->getAssetModelClassName();
+        eval(<<<PHP
```

Comment:
note, that I know there is some existing code (much light tho) in the current generic asset implementation.
I ticked on the length of the eval below.

This is not a mandatory thing, and I searched without finding a convincing solution about this, but maybe we can have some abstract classes implementing the code, and use `class_alias` to rename them

Reply:
I already looked into that, but for slightly different reasons. I wondered if I could remove all eval uses in this class because the function is considered inherently unsafe and may be blocked by hosting providers/security teams.

The issue is that while you can reference a class by the alias, when you get the class of an object of that class you get the non-aliased name.

So we could have places where the "common/abstract" class name is saved or used and you lose the important information.

When I was testing with the main asset class, I tried using an anonymous class which would get aliased to the name we expect since you can't name anonymous classes. I found it would save the asset class name as the randomized one assigned to the anonymous class.

Reply:
hum, that's a shame.
I don't have more suggestions on that, I took like 2 hours trying to find something on the subject without much success.
I found something around class.serializable, but it seems to be a bit complex.

Reply:
> because the function is considered inherently unsafe and may be blocked by hosting providers/security teams.

As far as I know, `eval` is not technically a function and it cannot  be disabled using the `disable_functions` PHP config entry.

> I don't have more suggestions on that, I took like 2 hours trying to find something on the subject without much success.

I tried many solutions, and none of them was working as expected. Anyway, unless you use `static` properties/variables, all of the code could be placed in an abstract class that you can extend. Your eval block would be much smaller.

Reply:
I discussed about it yesterday with @orthagh. We could keep `eval` instructions, but we have to make them as light as possible. For this specific case, you can move all the declared methods in an abstract class (well, it cannot reaally be abstract because `RuleDictionnaryDropdown` / `Rule` are not abstract, but you can add some PHPDoc to explain that it is supposed to be considered as abstract).
The only thing you have to do is to be sue that `static` properties are declared in your concrete class, because if they are declared in the "abstract" class, then their values will be shared by all concrete classes.

Reply:
Same remark for all the new `eval` blocks.

Reply:
> As far as I know, `eval` is not technically a function and it cannot be disabled using the `disable_functions` PHP config entry.

Apparently so. Although, it doesn't mean web hosts still can't block it with something like `Suhosin` which is basically a PHP extension and a PHP source code patch. Having this blocked may be less of an issue than I initially believed.

Reply:
`Suhosin` is not maintained since a long time ago. Honestly, I do not think we should consider that `eval` could be disabled in the systems we want to support. By the way, some of the `Symfony` component we use are also using `eval` in edge cases, so I guess that, even if we find a solmution to not use `eval` in our own code, GLPI would still not work as exepected on these systems.

Reply:
I pushed a commit to move the code to abstract classes.